problems/http_test.go

```diff
@@ -0,0 +1,91 @@
+package problems_test
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opencensus.io/trace"
+
+	"github.com/SKF/go-rest-utility/problems"
+)
+
+func runWriteResponse(err error, correlationID uint64) *http.Response {
+	spanCtx := trace.SpanContext{}
+	binary.BigEndian.PutUint64(spanCtx.TraceID[8:], correlationID)
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		ctx, span := trace.StartSpanWithRemoteParent(r.Context(), "Handler", spanCtx)
+		defer span.End()
+
+		problems.WriteResponse(ctx, err, w, r)
+	}
+
+	r := httptest.NewRequest("GET", "http://example.com/", nil)
+	w := httptest.NewRecorder()
+
+	handler(w, r)
+
+	return w.Result()
+}
+
+func TestWriteResponse_VanillaError(t *testing.T) {
+	var (
+		actualError                = fmt.Errorf("hello")
+		actualCorrelationID uint64 = 0xDEADBEEF
+	)
+
+	resp := runWriteResponse(actualError, actualCorrelationID)
+	defer resp.Body.Close()
+
+	require.Equal(t, http.StatusInternalServerError, resp.StatusCode)
+
+	var problem problems.BasicProblem
+	if err := json.NewDecoder(resp.Body).Decode(&problem); err != nil {
+		require.NoError(t, err)
+	}
+
+	require.Equal(t, "/problems/internal-server-error", problem.Type)
+	require.Equal(t, "http://example.com/", problem.Instance)
+	require.Equal(t, "3735928559", problem.CorrelationID)
+}
+
+type ImportantProblem struct {
+	problems.BasicProblem
+
+	Important string
+}
+
+func TestWriteResponse_DecoratableProblem(t *testing.T) {
+	var (
+		actualError = ImportantProblem{
+			BasicProblem: problems.BasicProblem{
+				Type:   "/problems/custom",
+				Title:  "Custom Problem.",
+				Status: http.StatusTeapot,
+				Detail: "I'm very important!",
+			},
+			Important: "Very!",
+		}
+		actualCorrelationID uint64 = 0xDEADBEEF
```

Comment:
Why hex here but dec in assert?

Reply:
Because its gets converted on the Datadog side and in all our handlers (go-utility/log, WriteProblems). I guess we can assert the base 10 int value in the tests and convert it before inserting it into the trace?

Reply:
Or maybe I can just use the base 10 number here I guess....

Reply:
one way is.
corrID := 23423984

struct problem {
...
actualCorrID = corrID 
}

...
require.Equal(itoa(corrID, actualID)